src/HotChocolate/Core/src/Execution/Processing/ArgumentValue.cs

```diff
@@ -4,17 +4,19 @@
 
 namespace HotChocolate.Execution.Processing;
 
-public sealed class ArgumentValue
+public sealed class ArgumentValue : IInputFieldInfo
```

Comment:
The `ArgumentValue` is now a `IInputFieldInfo`.

The `IInputFieldInfo` interface provides access to the most important properties of an input field. With this, we can now substitute an actual field with a made-up field in core helpers like the `InputParser`. 

In our concrete case, we can preserve the resolver arguments allowing us to rewrite the schema without the need for the resolver aligning to this new schema. The resolver in this case can still act against the original schema definition.

Reply:
Can we put this inside the code with some documentation?